### ISSUE TITLE
After YOLO11 OBB finetuning train images return None boxes

### ISSUE THREAD

**Issue by gabrielecaddeo** [ultralytics/ultralytics#19281]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi!
I am finetuning YOLO OBB on a small (around 100 images) dataset on 2 new classes (locks, keyholes). I am using a small amount of images just to understand if YOLO can overfit on those data. Since I know that the number of images can be low, I am testing the approach on a train image. 

It seems that the model converge, but when I test on a train image I get ```None``` boxes. Here's my script:
```python 
from ultralytics import YOLO
import wandb


# Load a pretrained YOLOv9 model
model = YOLO("yolo11n-obb.pt")
model_path = "yolo_rot_lock1_2.pt"

exp_name = "yolorotations"
wandb.login(key=my_key)
wandb.init(
    project="my_project",
    entity='my_entity',
    group='yolo_training',
    name=exp_name)
# Train the model on your dataset
results = model.train(
    data="./data.yaml",
    epochs=100,
    project="yolo_training",  # wandb project name
    name="yolo11rot"     # wandb run name
)
model = YOLO('./yolo_training/yolo11rot/weights/best.pt')
results = model.predict(source='./rotation_lock1/data/images/train/frame_00000140_left_00.png', conf=0.25)

print(results[0].boxes)
model.save(model_path)
wandb.log_model(model_path)
wandb.finish()
```
I already tried with a lower conf, e.g. ```conf=0.1```. The same approach worked for non-OBB YOLO, with the same images but with non-rotated boxes, so less precise boxes.
I thought that this problem could arise from the labels, but I followed the documentation and the results in the training were good (as it is possible to see in the attached image), so I do not think it was a problem of labels. Moreover, I am not specifying the ```imgsz``` parameter, so I am keeping the default one. What do you think could be the problem? 
Thanks

![Image](https://github.com/user-attachments/assets/4c8b6549-e06e-451b-ba94-205b669dc674)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @gabrielecaddeo, thank you for your interest in Ultralytics 🚀! We appreciate you bringing this up. Let's work together to dig into the issue.

If this is a 🐛 Bug Report, it would be very helpful if you could provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that demonstrates the problem. This helps us debug more efficiently. Based on your description, you’ve included some of the relevant code, which is great! If possible, please also share:
- A small subset of your dataset
- The `data.yaml` file you're using

For custom training ❓ Questions, make sure you've reviewed our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) and provide as much detail as possible (e.g., dataset examples, training logs, or unexpected behaviors). The more information we have, the better!

## Recommendations to Check:
1. **Dataset Labels:** You mentioned following the documentation, but double-checking your `.yaml` file and label formats (especially OBB formats) is crucial.
2. **Confidence Threshold:** You’ve already tried adjusting the `conf` parameter, which is great. Ensure you also test other inference parameters like `iou` as it might help.
3. **Image Dimensions:** While the default `imgsz` should work, sometimes explicitly setting it in the `train()` and `predict()` calls could provide more consistency.
4. **Upgrade Your Environment:** Ensure you are using the latest `ultralytics` version and dependencies. Upgrade by running:
```bash
pip install -U ultralytics
```

We also encourage collaboration with the growing Ultralytics community. Whether you’re debugging or brainstorming ideas, the following platforms are valuable resources:
- [Ultralytics Docs](https://docs.ultralytics.com/) 📚
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com/) 🗨️
- [Subreddit](https://reddit.com/r/Ultralytics) 📢

## Verified Environments

If you haven't already, you might find it worth testing in any of these verified, pre-configured environments:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Continuous Integration Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If the badge above shows green, the latest tests across all YOLO tasks are passing in Windows, macOS, and Ubuntu environments.

This is an automated response to assist you. An Ultralytics engineer will also review your issue and get back to you as soon as possible. Let us know if you have more details or updates to add in the meantime! 😊

**Comment by gabrielecaddeo** [comment]
Closing the issue, thanks

**Comment by UltralyticsAssistant** [comment]
The potential fix for this issue has been merged into the main branch via [PR #19282](https://github.com/ultralytics/ultralytics/pull/19282)! 🚀 Huge thanks to the author @Y-T-G and contributor @Laughing-q for their valuable efforts in improving the project. 🛠️🎉

### 🔧 Key Changes:
The PR enhances documentation for interpreting and using prediction results across multiple tasks, including:
- **Detection Results**: Clear examples for handling box coordinates (xywh, xyxy), class names, and confidence scores.
- **Oriented Bounding Boxes (OBBs)**: Guidance on accessing center, dimensions, rotation (xywhr), polygon data, and metadata.
- **Pose Estimation**: Clarified instructions for extracting keypoint coordinates, normalized keypoints, and visibility data.
- **Segmentation**: Expanded examples for retrieving masks in polygon and matrix formats, including normalization options.

These updates aim to reduce developer confusion, streamline workflows, and make results interpretation more beginner-friendly. We encourage testing the changes and sharing feedback!

### 🧪 How to Test:
You can test the latest changes by installing the updated version from the main branch:
```bash
pip install git+https://github.com/ultralytics/ultralytics.git@main
```
Alternatively, you can wait for the next official PyPI release to get these updates.

### 📢 Feedback Request:
Please let us know if this resolves the issue or if further adjustments are needed. Your feedback helps us refine and improve Ultralytics for everyone! 

Thank you again for reporting this issue and for your contributions to the community. 🙏 Feel free to reach out if the problem persists or if there’s anything else we can help with! 🚀 Happy coding!
